test/spec/index.js

```diff
@@ -0,0 +1,212 @@
+var assert = require('chai').assert;
+var standardComponents = require('../../src/scripts/components');
+var sinon = require('sinon');
+
+/**
+ * @return {void}
+ */
+function stubPebble() {
+  global.Pebble = {
+    addEventListener: sinon.stub(),
+    openURL: sinon.stub(),
+    sendAppMessage: sinon.stub()
+  };
+}
+
+describe('Clay', function() {
+  describe('Clay constructor', function() {
+    it('throws if the config is not an array', function() {
+      assert.throws(function() {
+        fixture.clay({});
+      }, /must be an Array/i);
+    });
+
+    it('throws if customFn is not a function', function() {
+      assert.throws(function() {
+        fixture.clay([], {});
+      }, /must be a function/i);
+    });
+
+    it('does not throw if customFn is undefined or a function', function() {
+      assert.doesNotThrow(function() {
+        fixture.clay([], function() {});
+      });
+
+      assert.doesNotThrow(function() {
+        fixture.clay([]).customFn();
+      });
+    });
+
+    it('registers the standard components present in the config', function() {
+      var config = fixture.config(
+        ['input', 'input', 'select', 'custom', ['color']],
+        false
+      );
+      var clay = fixture.clay(config);
+      assert.deepEqual(clay.components, {
+        input: standardComponents['input'],
+        select: standardComponents['select'],
+        color: standardComponents['color']
+      });
+    });
+
+    it('handles the "showConfiguration" event if autoHandleEvents is not false',
+    function() {
+      stubPebble();
+      var clay = fixture.clay([]);
+      Pebble.addEventListener.withArgs('showConfiguration').callArg(1);
+
+      assert(Pebble.addEventListener.calledWith('showConfiguration'));
+      assert(Pebble.openURL.calledWith(clay.generateUrl()));
+    });
+
+    it('handles the "showConfiguration" event if autoHandleEvents is not false',
+    function() {
+      stubPebble();
+      fixture.clay([]);
+      var logStub = sinon.stub(console, 'log');
+      Pebble.addEventListener
+        .withArgs('webviewclosed')
+        .callArgWith(1, { response: '%7B%22appKey%22%3A%22value%22%7D' });
+
+      assert(Pebble.addEventListener.calledWith('webviewclosed'));
+      assert(Pebble.sendAppMessage.calledWith({ appKey: 'value' }));
+
+      Pebble.sendAppMessage.callArg(1);
+      assert(logStub.calledWith('Sent config data to Pebble'));
+
+      Pebble.sendAppMessage.callArgWith(2, {some: 'error'});
+      assert(logStub.calledWith('Failed to send config data!'));
+      assert(logStub.calledWith('{"some":"error"}'));
+
+      logStub.restore();
+    });
+
+    it('handles an empty response in the "webviewclosed" handler', function() {
+      stubPebble();
+      fixture.clay([]);
+      Pebble.addEventListener.withArgs('webviewclosed').callArgWith(1, undefined);
+
+      assert(Pebble.addEventListener.calledWith('webviewclosed'));
+      assert.strictEqual(Pebble.sendAppMessage.callCount, 0);
+    });
+  });
```

Comment:
Would be good to have a test that verifies that it doesn't handle the events if told not to.